### PR TITLE
Add tags for ingredient

### LINKED ISSUE #49: Missing tag function for Ingredient
As an athlete who does meal preparation, I want to tag my ingredients as based on its main nutrient (E.G. Vitamin C, Fats) so that I make my recipe based on the these categories instead.

### PR DESCRIPTION
**Pull request by Trav1sT**
closes #49

### CODE REVIEW

File: src/main/java/chopchop/logic/parser/commands/AddCommandParser.java [hunk 1]
```diff
@@ -202,12 +207,12 @@ private static AddRecipeCommand createAddRecipeCommand(String name,
     }
 
 
-    private static AddIngredientCommand createAddIngredientCommand(String name,
-        Optional<Quantity> qty, Optional<String> expiry) {
+    private static AddIngredientCommand createAddIngredientCommand(String name, Optional<Quantity> qty,
+                                   Optional<String> expiry, Optional<Set<String>> tags) {
```
Comment (zhiayang) [inline src/main/java/chopchop/logic/parser/commands/AddCommandParser.java:211]: what's with this monkaMEGA indentation

File: src/main/java/chopchop/model/attributes/Tag.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Represents a Tag for both Recipe and Ingredient.
+ * E.G. "Halal", "Vegetarian" for Recipe. "Carbs", "Vitamin E" for Ingredient.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidTagName(String)}.
+ */
+public class Tag {
+
+    public static final String MESSAGE_CONSTRAINTS = "Tag name should be alphanumeric";
+    public static final String VALIDATION_REGEX = "\\p{Alnum}+";
```
Comment (zhiayang) [inline src/main/java/chopchop/model/attributes/Tag.java:14]: doesn't this regex mean tag names can't contain spaces?

File: src/main/java/chopchop/model/attributes/Tag.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    public String getTagName() {
+        return this.tagName;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+            || (other instanceof Tag // instanceof handles nulls
+            && tagName.equals(((Tag) other).tagName)); // state check
```
Comment (zhiayang) [inline src/main/java/chopchop/model/attributes/Tag.java:44]: eh i dun understand why y'all love to copy-paste this equals method with these same useless comments

File: src/main/java/chopchop/model/ingredient/Ingredient.java [hunk 4]
```diff
@@ -95,6 +105,26 @@ public Quantity getQuantity() {
...
+    public String getTagList() {
+        if (this.tags.isEmpty()) {
+            return "No tags attached";
+        }
+        StringBuilder sb = new StringBuilder();
+        int index = 1;
+        for (var tag : this.tags) {
+            sb.append(index)
+                .append(" : ")
+                .append(tag.toString().substring(1, tag.toString().length() - 1))
```
Comment (zhiayang) [inline src/main/java/chopchop/model/ingredient/Ingredient.java:121]: did you forget that you had `Tag::getTagName()`

**Comment by fall9x** [comment]
Looks good to merge!

**Comment by fall9x** [comment]
Looks good to merge!

**Pull request merged**